client/src/services/api/hooks/codeReview/codeReviewUtils.ts

```diff
@@ -45,5 +53,103 @@ export class CodeReviewMetricsUtils {
       pullRequests,
       errorCount: data.errorCount,
     };
-  };
+  }
+
+  static getFilteredPullRequestsByReviewers(
+    data: Omit<IFetchedRawCodeReviewResponse, "count" | "filteredCount"> | undefined,
+    filters: IContextClientFilterSquad[],
+  ): IFetchedCodeReviewResponse | undefined {
+    if (!data) {
+      return;
+    }
+
+    const { reviewersMap } = this.#getSquadsUserIdsMap(filters);
+
+    const pullRequests = data.pullRequests.map((pullRequest) => {
+      const filteredVotesTimeline = this.#filterVotesTimelineByReviewers(pullRequest.votesTimeline, reviewersMap);
+      const votes = this.#getPullRequestVotes(filteredVotesTimeline);
+
+      const filteredVotesHistoryTimeline = this.#filterVotesTimelineByReviewers(
+        pullRequest.votesHistoryTimeline,
+        reviewersMap,
+      );
+      const votesHistory = this.#getPullRequestVotes(filteredVotesHistoryTimeline);
+
+      const filteredThreads = this.#getFilteredThreads(pullRequest.threads, reviewersMap, pullRequest.authorId);
+
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { threads, ...rest } = pullRequest;
```

Comment:
@sathish-thangavel will this change works? as I need to omit threads while spreading, I used the above operation but Eslint is throwing no unused vars, so I suppressed it.